syntax = "proto3";
package lobby;

// Define the different types of actions.
enum LobbyAction {
  UNKNOWN = 0;
  JOIN_LOBBY = 1;
}

enum GameType {
  NONE = 0;
  TEXAS = 1;
  FIVE_HAND = 2;
}

// Define the structure of a game history entry
message GameHistoryEntry {
  int32 game_id = 1;
  string game_type = 2;
  string date = 3;
  int32 money_won = 4;
}


message LobbyRequest {
  // FOR CLIENT USE


  // Define the action from enum
  LobbyAction action = 1;

  // username + passhash used for creating an deleting accounts
  string username = 2;
  string passhash = 3;

  int32 money_to_add = 4;
  int32 game_type = 5;
}

message LobbyResponse {
  LobbyAction action = 1;

  // for client use (success vs failure)
  bool result = 2;

  // get lobby to join
  string game_lobby = 3;

  // get game history
  repeated GameHistoryEntry game_history = 4;

  int32 moolah = 5;
}

service LobbyService {
  // A single bidirectional stream for all lobby operations
  rpc Lobby(stream LobbyRequest) returns (stream LobbyResponse);
}